pynxtools/dataconverter/readers/xrd/xrd_helper.py

```diff
@@ -0,0 +1,244 @@
+                    # data or field val
+                    if val_atr_key == 'value':
+                        template[nx_key] = xrd_dict.get(path, None)
+                    # attr e.g. @units
+                    elif path and val_atr_key.startswith('@'):
+                        template[nx_key + '/' + val_atr_key] = xrd_dict.get(path, None)
+            if not isinstance(val, dict) and isinstance(val, str):
+                template[nx_key] = val
+
+    fill_template_from_config_data(config_dict, template,
+                                   xrd_dict, file_term)
+
+    def two_theta_plot():
+        two_theta_gr = "/ENTRY[entry]/2theta_plot"
+        template[two_theta_gr + "/" + "@axes"] = ["two_theta"]
+        template[two_theta_gr + "/" + "@signal"] = "intensity"
+
+    def q_plot():
+        q_plot_gr = "/ENTRY[entry]/q_plot"
+        alpha_2 = template.get("/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_two",
+                               None)
+        alpha_1 = template.get("/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_one",
+                               None)
+        two_theta: np.ndarray = template.get("/ENTRY[entry]/2theta_plot/two_theta", None)
+        if isinstance(two_theta, np.ndarray):
+            theta: np.ndarray = two_theta / 2
+        # pylint: disable=line-too-long
+        ratio_key = "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/ratio_k_alphatwo_k_alphaone"
+        if alpha_1 and alpha_2:
+            ratio = alpha_2 / alpha_1
+            template[ratio_key] = ratio
+            lamda = ratio * alpha_1 + (1 - ratio) * alpha_2
+            q_vec = (4 * np.pi / lamda) * np.sin(np.deg2rad(theta))
+            template[q_plot_gr + "/" + "q_vec"] = q_vec
+            template[q_plot_gr + "/" + "@q_vec_indicies"] = 0
+            template[q_plot_gr + "/" + "@axes"] = ["q_vec"]
+
+    def handle_special_fields():
+        """Some fields need special treatment."""
+
+        key = "/ENTRY[entry]/COLLECTION[collection]/goniometer_x"
+        gonio_x = template.get(key, None)
+
+        template[key] = gonio_x[0] if (isinstance(gonio_x, np.ndarray)
+                                       and gonio_x.shape == (1,)) else gonio_x
+
+        key = "/ENTRY[entry]/COLLECTION[collection]/goniometer_y"
+        gonio_y = template.get(key, None)
+
+        template[key] = gonio_y[0] if (isinstance(gonio_y, np.ndarray)
+                                       and gonio_y.shape == (1,)) else gonio_y
+
+        key = "/ENTRY[entry]/COLLECTION[collection]/goniometer_z"
+        gonio_z = template.get(key, None)
+
+        template[key] = gonio_z[0] if (isinstance(gonio_z, np.ndarray)
+                                       and gonio_z.shape == (1,)) else gonio_z
+
+        key = "/ENTRY[entry]/COLLECTION[collection]/count_time"
+        count_time = template.get(key, None)
+
+        template[count_time] = count_time[0] if (isinstance(count_time, np.ndarray)
```

Comment:
This should be template[key] instead of template[count_time]

Reply:
fixed!